src/structure_managers/structure_manager.hh

```diff
@@ -419,77 +469,74 @@ namespace rascal {
-      this->template validate_property_t<UserProperty_t>(property);
-      return std::static_pointer_cast<UserProperty_t>(property);
-    }
-
-    void register_property(std::shared_ptr<PropertyBase> property,
-                           const std::string & name) {
-      this->properties[name] = property;
-    }
     /**
-     * Get a property of a given name. Create it if it does not exist.
+     * Returns a typed property of the given name.
      *
-     * @tparam UserProperty_t full type of the property to return
+     * @tparam UserProperty_t full type of the property to return.
      *
-     * @param name name of the property to get
+     * @param name the name of the property to get.
+     * @param validate_property property is validated if this parameter is true,
+     * see validate_property_t
+     * @param force_creation if the property does not exist in the manager stack
+     * the property is created and returned. The validation step is skipped in
+     * this case.
      *
-     * @throw runtime_error if UserProperty_t is not compatible with property
-     * of the given name
+     * @throw runtime_error If validate_property is true and UserProperty_t is
+     * not compatible with the property with the given name.
+     * @throw runtime_error If force_creation is false and property has not been
+     * found in manager stack.
      */
     template <typename UserProperty_t>
-    std::shared_ptr<UserProperty_t> get_property_ptr(const std::string & name) {
-      if (this->has_property(name)) {
-        auto property{this->get_property(name)};
-        UserProperty_t::check_compatibility(*property);
-        return std::static_pointer_cast<UserProperty_t>(property);
-      } else {
-        auto property{std::make_shared<UserProperty_t>(this->implementation())};
-        this->register_property(property, name);
-        return property;
+    std::shared_ptr<UserProperty_t> get_property_ptr(const std::string & name,
+                                                     bool validate_property,
+                                                     bool force_creation) {
+      if (this->has_stack_property(name)) {
+        return this->template get_property_ptr<UserProperty_t>(
+            name, validate_property);
+      }
+      if (force_creation) {
+        this->template create_property<UserProperty_t>(name);
+        return this->template get_property_ptr<UserProperty_t>(name, false);
       }
+      std::stringstream error{};
+      error << "No property of name '" << name << "' has been registered";
+      throw std::runtime_error(error.str());
     }
 
     template <typename UserProperty_t>
-    UserProperty_t & get_property_ref(const std::string & name) {
-      return *this->template get_property_ptr<UserProperty_t>(name);
+    UserProperty_t & get_property_ref(const std::string & name,
+                                      bool validate_property,
+                                      bool force_creation) {
```

Comment:
I would make it optional.
```suggestion
                                      bool force_creation = false) {
```

Reply:
force creation and validate property is now optional, and by default force creation is false and validate is true